examples-next/auth/schema.ts

```diff
@@ -0,0 +1,144 @@
+import { createSchema, list } from '@keystone-spike/keystone/schema';
+import { text, checkbox, password } from '@keystone-spike/fields';
+
+/**
+ * TODO: with the new session API, we're breaking keystone's current assumptions about the
+ * `authentication` argument provided to access control functions. This will be a breaking change
+ * for developers using Keystone.
+ *
+ * This needs to be implemented (replace `authentication` with `session`) before the code below
+ * will work.
+ *
+ * While we're here I'd like to propose some other changes to improve the ergonomics; see the types
+ * below for my proposal. Note that I'm specifically omitting the graphql context and any way of
+ * executing queries because I'd like to review that API as well, separately (and we can add it in
+ * later). This may be a bit contentious, but we need a new object anyway so I'm happy if in the
+ * first pass we just implement ({ session, item }) argument so the code below works and we can
+ * discuss the rest of the proposal as a group.
+ *
+ * Note that custom schema access control deviates significantly from the list and field access
+ * control arguments; we need to revisit that in the context of how graphql schema extensions work
+ * now. See https://www.keystonejs.com/api/access-control#custom-schema-access-control
+ *
+ * Also note, I'm not sure the reasoning behind why ListAccess seems to be resolved once per
+ * mutation for mutations that operate on multiple items, while FieldAccess seems to be resolved
+ * once per item (many times per mutation) for the same. At least, that's what I think is going on.
+ *
+ * This seems like a real foot-gun, and I'd prefer that they're implemented consistently, in favor
+ * of aligning on FieldAccessInput and resolving the function once per item being updated in the
+ * case of mutations that operate on multiple items.
+ *
+ * Or maybe my understanding is wrong? This is what's implied by the documentation (see
+ * https://www.keystonejs.com/api/access-control#field-level-access-control) but then if it's not
+ * the case I don't see how `existingItem` works in mutations that operate on multiple items.
```

Comment:
The API  here should be `existingItems` (note the plural), which  then allows looking at the items to determine if access is allowed.

Reply:
Yeah I'm not sure (haven't checked yet) if that's a bug or the docs are wrong... @mitchellhamilton did say that he thought the field access control is resolved differently to list access control but we want to check with @timleslie about it

Reply:
A quick search on GitHub turned up this: https://github.com/keystonejs/keystone/pull/3241/files / https://github.com/keystonejs/keystone/blob/5f2689dc1acb8b7299961f6ffef72b218f52cc3e/packages/keystone/lib/ListTypes/list.js#L814-L839

I was wrong in my assertion that it's a plural. It's still singular, but gives an array of things to check by the looks of it.